src/Controls/tests/DeviceTests/Elements/Shell/ShellTests.Windows.cs

```diff
@@ -27,6 +28,43 @@ protected Task CheckFlyoutState(ShellHandler handler, bool desiredState)
 			return Task.CompletedTask;
 		}
 
+		[Fact(DisplayName = "Shell FlyoutIcon Initializes Correctly")]
+		public async Task ShellFlyoutIconInitializesCorrectly()
+		{
+			SetupBuilder();
+
+			var shell = await CreateShellAsync((shell) =>
+			{
+				shell.FlyoutBehavior = FlyoutBehavior.Flyout;
+				shell.FlyoutIcon = "red.png";
+
+				var shellItem = new FlyoutItem();
+				shellItem.Items.Add(new ContentPage());
+				shell.Items.Add(shellItem);
+			});
+
+			await InvokeOnMainThreadAsync(async () =>
```

Comment:
`CreateHandleAndAddToWindow` already invokes on the main thread so you can remove this part.

Reply:
Right, removed.